onnxscript/function_libs/torch_lib/ops/nn.py

```diff
@@ -1714,10 +1714,12 @@ def _causal_attention_mask(query: TFloat, key: TFloat) -> TFloat:
     # }
     attn_mask = op.Trilu(attn_mask, upper=0)
     # The causal mask has 0s in the lower triangle and -inf in the upper triangle.
+    zero = op.CastLike(op.Constant(value_float=0.0), attn_mask)
```

Comment:
attn_mask  is created from line 1713, which should always be float at this point?